libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -852,11 +852,7 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
 
 - (void)restartAuthentication {
     @synchronized (self.authBlockList) {
-        if (!self.authenticating) {
-            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
-            return;
-        }
-        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self log:SFLogLevelInfo format:@"%@: Restarting authentication process.", NSStringFromSelector(_cmd)];
```

Comment:
And as I mentioned in my first comment, because the "Cancel the initial alert" use case cancels the in-progress authentication, it breaks the contract created by the consumer.

With every other action of our asynchronous public login methods, there are only two paths out: completion or failure.  And those run through the respective handler blocks passed into those methods. `self.authenticating` is just a convenience property to determine whether there have been auth callback container object(s) established.  No auth callback containers means the login* methods either haven't been called, or have completed.

In the event of a successful "restarted" authentication in this use case, what's the tie-back to the original caller?  There isn't one.  Broken contract.

If this is an opportunity to close the gap on that aberration, I think we should do it—in addition to the fact that I still don't see any further value to having the alert anyway.